adding textures to each segment

CarlosRA97 opened the pull request:
It enables the user to use image shaders by painting each segment with an image

CarlosRA97 committed:
adding textures to each segment

apgapg commented:
THanks @CarlosRA97 for this PR.
Looks Okay to me, but i never encountered any textures.
Can you explain a bit more? or attach any screenshots?

CarlosRA97 commented:
Something like this
![IMG_20230427_173609_105.jpg](https://user-images.githubusercontent.com/6619716/234917889-1e9f429b-0053-438a-a6e2-e20c57576438.jpg)

For example in the case of a flavor refiller with that you can see visually how much of what flavor, you're going to get

Or for a business how much of which product had sell in that month and check it in a glance

Thanks for your library

apgapg commented:
Okay looks great

apgapg commented:
Can you also help me update README with this new feature?

CarlosRA97 committed:
adding documentation and an example of textures

CarlosRA97 committed:
adding image example

CarlosRA97 committed:
fix on extension of image sample

CarlosRA97 commented:
Yes, I already commited the new feature in README.md

CarlosRA97 committed:
Update pie_chart.dart

CarlosRA97 committed:
Update chart_painter.dart

CarlosRA97 committed:
fix: shouldRepaint ignored actual segment values, causing stale/janky redraws

_totalAngle only reflects the constant full-circle sweep (frozen animFraction
* static totalDegrees), never the per-segment values, so shouldRepaint almost
always returned false after the first frame. Real dataMap updates only
appeared on screen when something else incidentally forced a relayout.
